ivadomed/models.py

```diff
@@ -425,14 +428,36 @@ def __init__(self, out_channel=1, depth=3, drop_rate=0.4, bn_momentum=0.1,
         self.last_conv = conv(in_channel // 2, out_channel, kernel_size=3, padding=1)
         self.last_film = FiLMlayer(n_metadata, self.out_channel) if film_layers and film_layers[-1] else None
         self.softmax = nn.Softmax(dim=1)
+        
+        ### ATTENTION MODULE ###
+        if self.attention:
+            
+            self.gating = UnetGridGatingSignal2(self.base_n_filter * 2**(self.depth),
```

Comment:
Decoder has an attribute `is_2d` if this attribute is set to false, we expect to use a 3D version of the UNet. Here,  `UnetGridGatingSignal2` should be `UnetGridGatingSignal3` if `is_2d` is set to False.